ams/server/src/main/java/com/netease/arctic/server/table/TableRuntime.java

```diff
@@ -208,10 +228,10 @@ public void cleanPendingInput() {
     invokeConsisitency(
         () -> {
           pendingInput = null;
-          if (optimizingStatus == OptimizingStatus.PENDING) {
+          if (optimizingStatus == OptimizingStatus.PLANNING) {
```

Comment:
Why pending status not handled?

Reply:
I see, this method is somehow coupled with beginPlanning

Reply:
In fact, the previous `pending` is split into two states: `pending` and `planning` , so the judgment conditions here need to be updated.

Reply:
ye~

I mean what if current status is pending?
We know that there's no such situation here because this method is only called after beginPlanning. That's why I say there's a couple with beginPlanning, which could be decoupled. 

How about adding pending status logic here. Image some day we don't always call clearPendingInput after beingPlanning.

Reply:
> mean what if current status is pending? We know that there's no such situation here because this method is only called after beginPlanning. That's why I say there's a couple with beginPlanning, which could be decoupled.
> 
> How about adding pending status logic here. Image some day we don't always call clearPendingInput after beingPlanning.

get.